platform/platform-filter-delegate/src/main/java/org/codice/ddf/platform/filter/delegate/DelegateServletFilter.java

```diff
@@ -112,6 +102,168 @@ public void doFilter(ServletRequest servletRequest, ServletResponse servletRespo
+         * See the pax-web Whiteboard documentation and
+         * {@link HttpWhiteboardConstants#HTTP_WHITEBOARD_FILTER_NAME} for how to configure
+         * {@link Filter} services with a filter name.
+         */
+        @Override
+        public String getFilterName() {
+            final String filterNameFromTheServiceProperty =
+                    ServicePropertiesUtils.getStringProperty(securityFilterServiceReference,
+                            HttpWhiteboardConstants.HTTP_WHITEBOARD_FILTER_NAME);
+            // If this service property is not specified, the fully qualified name of the service object's class is used as the servlet filter name.
+            if (StringUtils.isBlank(filterNameFromTheServiceProperty)) {
+                return getContext().getService(securityFilterServiceReference)
+                        .getClass()
+                        .getCanonicalName();
+            } else {
+                return filterNameFromTheServiceProperty;
+            }
+        }
+
+        @Override
+        public ServletContext getServletContext() {
+            return servletContext;
+        }
+
+        @Override
+        public String getInitParameter(String name) {
+            return initParams.get(name);
+        }
+
+        @Override
+        public Enumeration<String> getInitParameterNames() {
+            return Collections.enumeration(initParams.keySet());
+        }
+
+        /**
+         * This code to create the init params from a {@link ServiceReference<Filter>} is copied
+         * from {@link org.ops4j.pax.web.extender.whiteboard.internal.tracker.ServletTracker#createWebElement(ServiceReference, javax.servlet.Servlet)}.
+         * See the pax-web Whiteboard documentation and
+         * {@link ExtenderConstants#PROPERTY_INIT_PREFIX} for how to configure {@link Filter}
+         * services with init params.
+         */
+        private Map<String, String> createInitParams(
+                final ServiceReference<SecurityFilter> securityFilterServiceReference) {
+            String[] initParamKeys = securityFilterServiceReference.getPropertyKeys();
+            String initPrefixProp = ServicePropertiesUtils.getStringProperty(
+                    securityFilterServiceReference,
+                    ExtenderConstants.PROPERTY_INIT_PREFIX);
+            if (initPrefixProp == null) {
+                initPrefixProp = ExtenderConstants.DEFAULT_INIT_PREFIX_PROP;
+            }
+
+            // make all the service parameters available as initParams
+            Map<String, String> initParams = new HashMap<>();
+            for (String key : initParamKeys) {
+                String value = securityFilterServiceReference.getProperty(key) == null ?
+                        "" :
+                        securityFilterServiceReference.getProperty(key)
+                                .toString();
+
+                // if the prefix is null or empty, match is true, otherwise its
```

Comment:
This code and comments are copied from [org.ops4j.pax.web.extender.whiteboard.internal.tracker.ServletTracker#createWebElement](https://github.com/ops4j/org.ops4j.pax.web/blob/web-6.0.3/pax-web-extender-whiteboard/src/main/java/org/ops4j/pax/web/extender/whiteboard/internal/tracker/ServletTracker.java#L76-L224).